components/datetime/src/pattern/reference/parser.rs

```diff
@@ -103,8 +103,7 @@ impl<'p> Parser<'p> {
                 (Segment::Literal { ref mut quoted, .. }, false) => {
                     *quoted = !*quoted;
                 }
-                #[allow(clippy::panic)] // TODO(#1668) Clippy exceptions need docs or fixing.
-                _ => panic!(),
+                _ => unreachable!("Generic pattern has no symbols."),
```

Comment:
Thought: I wonder if there's a way to make `Parser` more generic or something so that this can be enforced at compile time rather than requiring an `unreachable!`. But it's fine.